tests/test_pytorch_dataloader_api.py

```diff
@@ -0,0 +1,212 @@
+            with tiledb.DenseArray(tiledb_uri_x, mode="r") as x, tiledb.DenseArray(
+                tiledb_uri_y, mode="r"
+            ) as y:
+                with self.assertRaises(Exception):
+                    PyTorchTileDBDenseDataset(
+                        x_array=x, y_array=y, batch_size=BATCH_SIZE
+                    )
+
+    def test_no_duplicates_with_multiple_workers(self):
+        for workers in NUM_OF_WORKERS[2:]:
+            with self.subTest():
+                with tempfile.TemporaryDirectory() as tiledb_uri_x, tempfile.TemporaryDirectory() as tiledb_uri_y:
+
+                    dataset_shape_x = (ROWS,) + INPUT_SHAPES[1]
+                    dataset_shape_y = (ROWS,)
+
+                    ingest_in_tiledb(
+                        uri=tiledb_uri_x,
+                        data=np.random.rand(*dataset_shape_x),
+                        batch_size=BATCH_SIZE,
+                    )
+                    ingest_in_tiledb(
+                        uri=tiledb_uri_y,
+                        data=np.random.randint(
+                            low=0, high=NUM_OF_CLASSES, size=dataset_shape_y
+                        ),
+                        batch_size=BATCH_SIZE,
+                    )
+
+                    with tiledb.DenseArray(
+                        tiledb_uri_x, mode="r"
+                    ) as x, tiledb.DenseArray(tiledb_uri_y, mode="r") as y:
+
+                        tiledb_dataset = PyTorchTileDBDenseDataset(
+                            x_array=x, y_array=y, batch_size=BATCH_SIZE
+                        )
+
+                        self.assertIsInstance(
+                            tiledb_dataset, torch.utils.data.IterableDataset
+                        )
+
+                        train_loader = torch.utils.data.DataLoader(
+                            tiledb_dataset, batch_size=None, num_workers=workers
+                        )
+
+                        unique_inputs = []
+                        unique_labels = []
+
+                        for batchindx, data in enumerate(train_loader):
+                            # Keep unique X tensors
+                            if not any(
+                                np.array_equal(data[0].numpy(), unique_input)
+                                for unique_input in unique_inputs
+                            ):
+                                unique_inputs.append(data[0].numpy())
+
+                            # Keep unique Y tensors
+                            if not any(
+                                np.array_equal(data[1].numpy(), unique_label)
+                                for unique_label in unique_labels
+                            ):
+                                unique_labels.append(data[1].numpy())
+
+                        self.assertEqual(len(unique_labels), len(unique_inputs))
+                        self.assertEqual(len(unique_inputs) - 1, batchindx)
+                        self.assertEqual(len(unique_labels) - 1, batchindx)
```

Comment:
I have not research on workers yet, but would it make sense to have a test in case a worker receives empty batch so we check the behaviour there? Same principal of work sharing test with duplicates.

Reply:
I don't think that a worker will ever produce an empty batch, because the dataset is equally divided among the available workers.